openwisp-config/files/openwisp.agent

```diff
@@ -370,6 +372,25 @@ get_checksum() {
 		return 2
 	fi
 
+	if is_http_status "$1" 404; then
+		for attempt_no in $(seq 1 "$MAX_RETRIES"); do
+			sleep "$RETRY_DELAY"
+			$FETCH_COMMAND -i "$CHECKSUM_URL" >"$1"
+			if is_http_status "$1" 404; then
+				if [ "$attempt_no" -eq "$MAX_RETRIES" ]; then
+					local status
+					status=$(head -n 1 "$1")
+					logger -s "Checksum not found $status" \
+						-t openwisp \
+						-p daemon.err
+					exit 1
```

Comment:
Let's exit with zero to signal to the rest of the system that all is fine (otherwise the system will think the agent crashed).

```suggestion
					exit 0
```